src/map/mob.c

```diff
@@ -1464,7 +1464,7 @@ static int mob_ai_sub_hard_slavemob(struct mob_data *md, int64 tick)
 			mob_stop_attack(md);
 			const struct mob_data *m_md = BL_CCAST(BL_MOB, bl); // Can be NULL due to master being BL_PC
 			// If master is BL_MOB and in battle, lock & chase to master's target instead, unless configured not to.
-			if ((battle_config.slave_chase_masters_chasetarget == 0 || (m_md != NULL && !mob->is_in_battle_state(m_md)))
+			if ((bl->type == BL_PC || battle_config.slave_chase_masters_chasetarget == 0 || (m_md != NULL && !mob->is_in_battle_state(m_md)))
```

Comment:
slave should always follow master? or follow if only slave_chase_masters_chasetarget is 0?

Reply:
Slaves summoned by players should always follow their master.
Slaves summoned by monsters will follow either their master or the master's chase target, depending on `slave_chase_masters_chasetarget` battle flag and the situation. (See #2561.)